src/main/java/org/opentripplanner/updater/car_park/HslCarParkDataSource.java

```diff
@@ -0,0 +1,140 @@
+package org.opentripplanner.updater.car_park;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.vividsolutions.jts.geom.*;
+import org.opentripplanner.routing.car_park.CarPark;
+import org.opentripplanner.util.NonLocalizedString;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Load car parks from the HSL Park and Ride API.
+ *
+ * @author hannesj
+ */
+public class HslCarParkDataSource extends GenericJsonCarParkDataSource{
+
+    private static final Logger log = LoggerFactory.getLogger(HslCarParkDataSource.class);
+
+    private GeometryFactory gf = new GeometryFactory();
+
+    public HslCarParkDataSource() {
+        super("results");
+    }
+
+    public CarPark makeCarPark(JsonNode node) {
+        if (node.path("builtCapacity").path("CAR").isMissingNode()) return null;
+
+        CarPark station = new CarPark();
+        station.id = node.path("id").asText();
+        // TODO: add internalization
+        station.name = new NonLocalizedString(node.path("name").path("fi").asText());
+        try {
+            station.geometry = parseGeometry(node.path("location"));
+            station.y = station.geometry.getCentroid().getY();
+            station.x = station.geometry.getCentroid().getX();
+            station.realTimeData = false;
+            station.maxCapacity = node.path("builtCapacity").path("CAR").asInt();
+            if (!node.path("status").asText().equals("IN_OPERATION")) {
+                station.spacesAvailable = 0;
+            } else {
+                station.spacesAvailable = station.maxCapacity;
+            }
+            return station;
+        } catch (Exception e) {
+            log.warn("Error parsing car park " + station.id, e);
+            return null;
+        }
+    }
+
+
+    // TODO: These are inlined from GeometryDeserializer
```

Comment:
The input is valid GeoJSON geometry, so I think we should be prepared to parse it even though currently all of the values seem to be polygons.